### PR TITLE
Update documentation for batch scoring steps

### PR DESCRIPTION
**Pull request by jerrylian-db**
## What changes are proposed in this pull request?

As title. Note, as part of this PR, I decided to add `pipeline.yaml` links for the first mentioning of `pipeline.yaml` in each step, to make all the steps more consistent in how they format `pipeline.yaml`.

## How is this patch tested?

NA

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/regression`: Sklearn regression pipeline example
- [ ] `area/build`: Build and test infrastructure for MLflow pipeline example

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -236,7 +236,7 @@ The user-defined estimator function should be written in [`steps/train.py`](http
 and should return an unfitted estimator that is `sklearn`-compatible; that is, the returned object should define 
 `fit()` and `transform()` methods. `steps/train.py` contains an example placeholder function.
 
-The train step is configured by the `steps.train` section in pipeline.yaml:
+The train step is configured by the `steps.train` section in `pipeline.yaml`:
```
Comment (prithvikannan) [inline README.md]: can we add the link here like the other steps?

**Review by prithvikannan — APPROVED** [review]
Thanks for updating these docs! Left a small nit comment

File: pipeline.yaml [hunk 2]
```diff
@@ -80,18 +87,25 @@ steps:
     # be registered to the MLflow Model Registry
     allow_non_validated_model: false
   predict:
+    #
+    # FIXME::REQUIRED: Specify the output format of the scored data from predict
```
Comment (jinzhang21) [inline pipeline.yaml]: Can we make all batch scoring usage as optional? I don't think user has to define these before a model is trained.

File: profiles/databricks.yaml [hunk 3]
```diff
@@ -30,5 +30,5 @@ INGEST_DATA_LOCATION: ""
 # SCORED_OUTPUT_DATA_LOCATION: ""
 #
 # FIXME::OPTIONAL: Specify the format of the scored dataset. Natively supported formats are:
-#                  parquet, spark_sql, delta.
+#                  parquet, delta, table.
```
Comment (jinzhang21) [inline profiles/databricks.yaml:33]: Isn't `spark_sql` supported?

File: README.md [hunk 4]
```diff
@@ -316,6 +316,55 @@ Whether to allow registration of models that fail to meet performance thresholds
...
+trained model.
+
+#### Ingest Scoring step
+The ingest scoring step, defined in the `data_scoring` section in [`pipeline.yaml`](https://github.com/mlflow/mlp-regression-template/blob/main/pipeline.yaml), specifies the dataset used for batch scoring and has the same API as the [ingest step](#ingest-step).
+
+**Step artifacts**:
+- `ingested_scoring_data`: the ingested scoring data as a Pandas DataFrame.
+
+#### Predict step
+The 'predict' step uses the model registered by the [register step](#register-step) to score the ingested dataset produced by the [ingest scoring step](#ingest-scoring-step) and writes the resulting dataset to the specified output format and location. Namely, it uses the latest version of the registered model specified by the `model_name` attribute of the `pipeline.yaml` [register step](#register-step) definition. To fix a specific model for use in the 'predict' step, provide its model URI as the 'model_uri' attribute of the `pipeline.yaml` 'predict' step definition.
```
Comment (jinzhang21) [inline README.md]: Mind breaking down these really long lines into multiple lines for ease of read and review?

File: profiles/databricks.yaml (hunk 3 above)
Comment (jerrylian-db) [reply profiles/databricks.yaml:33]: No, it is not. We do not support writing with `spark_sql`

File: README.md (hunk 4 above)
Comment (jerrylian-db) [reply README.md]: Okay, will do!

File: profiles/databricks.yaml (hunk 3 above)
Comment (jinzhang21) [reply profiles/databricks.yaml:33]: Ah, yes. Nvm.

**Review by jinzhang21 — APPROVED** [review]

**Pull request merged**